document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/data/transaction/TransactionManager.java

```diff
@@ -76,8 +76,7 @@ public Transaction getTransaction(String transactionLink, String requestId) thro
      * @return a statically created rest template
      */
     private static RestTemplate createRestTemplate() {
-        HttpComponentsClientHttpRequestFactory requestFactory = new HttpComponentsClientHttpRequestFactory();
-        return new RestTemplate(requestFactory);
+        return new RestTemplate();
```

Comment:
Why are we opting for this default constructor over the [one that takes the factory?](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/client/RestTemplate.html#%3Cinit%3E(org.springframework.http.client.ClientHttpRequestFactory)) Does this not change the way the client behaves in some way?

Reply:
No It does not , We have discussed on call as well with you and Abhi, There was error  if request factory is used in constructor . So have to used default .

Reply:
I wonder if the error was due to an incorrect HttpComponents version being used as noted [on these docs](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/http/client/HttpComponentsClientHttpRequestFactory.html)

Additionally Spring docs actually suggest [RestClient](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/client/RestClient.html) now which might be a better option